flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -148,7 +149,10 @@
 	/** Service to contend for and retrieve the leadership of JM and RM */
 	private final HighAvailabilityServices highAvailabilityServices;
 
-	/** Blob cache manager used across jobs */
+	/** Blob server used across jobs */
+	private final BlobServer blobServer;
```

Comment:
I think the `JobMaster` should not depend on the `BlobServer` because the `BlobServer` might run on a different node (along side to the dispatcher, for example). Can we pass in the `BlobService` which can either be a `BlobServer` or a `BlobCache`? I think we have to move the `register/releaseJob` methods to the `BlobService` for that. Additionally, the `BlobServer` should simply do nothing when these methods are called. That way, we can also pass in a `BlobService` to the `Task` and the `TaskExecutor/TaskManager`